### PR TITLE
RATEST-293: Add platform upgrade set up wizard tests 

### PR DESCRIPTION
**Pull request by kdaud**

### CODE REVIEW

**Comment by dkayiwa** [comment]
What problem are you fixing with this?

**Comment by kdaud** [comment]
This [commit ](https://github.com/openmrs/openmrs-core/commit/b0d8a20291bc60abd1b3180302b8147f26921281) brings in the header page to display the text `Log in as a User that has the System Developer role to continue.` so have added an assertion method that confirms the text is displayed during the upgrade set up wizard.

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/UpgradeSteps.java [hunk 1]
```diff
@@ -29,7 +38,15 @@ public void finish() {
 		complete();
 	}
 
-	@Given("User enters credentials")
+	@Given("User is on login page")
+	public void visitLoginPage() {	
+		assertTrue(textExists(HEADER_TEXT));
+		assertTrue(textExists(SUPER_USER));
+		assertTrue(textExists(OPENMRS_VERSION));
```
Comment (dkayiwa) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/UpgradeSteps.java:45]: Will this test pass when the openmrs version is 3.x?

**Comment by dkayiwa** [comment]
Besides the upgrade, is the initial setup also tested with these changes?

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/UpgradeSteps.java (hunk 1 above)
Comment (kdaud) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/UpgradeSteps.java:45]: > Will this test pass when the openmrs version is 3.x?

no,

however, one challenge I got in testing out the version is that OpenMRS version evolves over time so we are currently in 2.x version and  when the platform version reaches 3.x then the test will be adjusted accordingly. 

@dkayiwa do you have a suggestion on the subject?

Comment (dkayiwa) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/UpgradeSteps.java:45]: @kdaud Why do you increase our maintenance burden?

Comment (kdaud) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/UpgradeSteps.java:45]: > @kdaud Why do you increase our maintenance burden?

@dkayiwa your question is too smart. Well, increasing the maintenance burden is not a good idea.

Comment (dkayiwa) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/UpgradeSteps.java:45]: I guess you can now come up with a smart solution! 😊

**Comment by dkayiwa** [comment]
So if i reverted any line of that commit, will these tests fail?

**Comment by kdaud** [comment]
> So if i reverted any line of that commit, will these tests fail?

Apart from this [line](https://github.com/openmrs/openmrs-core/blob/96e0ae83fc5c6da3521a99a27c43ddfb720b2e27/web/src/main/java/org/openmrs/web/filter/update/UpdateFilterModel.java#L37) which I find having the no effects on the UI, the rest of the lines when reverted the test fails. With the line mentioned above, it is supposed to show the footer on the page which is not true.

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/InitialSetupSteps.java [hunk 2]
```diff
@@ -15,6 +15,9 @@
 import org.junit.Assert;
 import org.openmrs.contrib.qaframework.helper.InitialSetupPage;
 import org.openmrs.contrib.qaframework.helper.TestBase;
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
```
Comment (dkayiwa) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/InitialSetupSteps.java:19]: Is this exception in use?

Comment (kdaud) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/InitialSetupSteps.java]: > Is this exception in use?

Have done the needful in the next subsequent commit with id `9779260`

**Comment by dkayiwa** [comment]
@kdaud i do not see the platform tests included in the above checks.

**Comment by kdaud** [comment]
> @kdaud i do not see the platform tests included in the above checks.

@dkayiwa all the platform tests were configured not to be triggered by any PR. However, they all run on every merge made on master. So that is the reason why they are not shown in the checks. What I often do for testing out the platform tests is to run a plan for each test against the branch via my fork.

**Comment by dkayiwa** [comment]
Are you doing any more work on this?

**Comment by kdaud** [comment]
If the ideal is changing their original set up(for not being triggered on the PR ) then that will call for additional config in their respective workflows to be triggered on every PR made on master.

**Comment by kdaud** [comment]
> Are you doing any more work on this?

We are good to go unless we are to change the original set up for platform tests ie making the tests be triggered on every PR!

**Comment by dkayiwa** [comment]
@kdaud do the platform tests run immediately on merging?

**Comment by kdaud** [comment]
> @kdaud do the platform tests run immediately on merging?

yes they do. take a look at the current master plan
Screenshot from 2022-02-15 16-18-38

**Comment by dkayiwa** [comment]
How about when i make changes to openmrs-core? Do they run immediately on commit?

**Comment by kdaud** [comment]
> How about when i make changes to openmrs-core? Do they run immediately on commit?

This is something am sure of, because what I know for now is that run every merge made on qaframework module.

**Comment by dkayiwa** [comment]
> am sure of

Is this a typo?

**Comment by kdaud** [comment]
> > am sure of
> 
> Is this a typo?



It is! I meant `am not sure of`


Am looking into whether they are triggered or not on a change made on the core, and will respond to you ASAP.

**Comment by dkayiwa** [comment]
I remember as if some one said they run once a day. I feel that this should be changed to every commit of openmrs-core

**Comment by kdaud** [comment]
> I remember as if some one said they run once a day.

I also think the person was pointing to the Bamboo plan that at-least run once every day. Usually this configured ci plan happens at 4:00 AM EAT.

> I feel that this should be changed to every commit of openmrs-core

This is a good idea, my understanding  on the subject is to be achieve the goal vai Bamboo config,

**Comment by dkayiwa** [comment]
How about the ones on this dashboard? https://github.com/openmrs/openmrs-contrib-qaframework

Will they immediately fail if i revert that commit in openmrs-core?

**Comment by kdaud** [comment]
> How about the ones on this dashboard? https://github.com/openmrs/openmrs-contrib-qaframework
> 
> Will they immediately fail if i revert that commit in openmrs-core?

AFAICT those on the dashboard won't fail until a merge is done on the qaframework repo to trigger them to run. This is when the tests will detect a change on the openmrs-core and this means a change done on the core is detected by the tests later.

**Comment by dkayiwa** [comment]
Can't you make them run immediately on openmrs-core commit?

**Comment by kdaud** [comment]
> Can't you make them run immediately on openmrs-core commit?

Have started off with the task!

**Pull request merged**